Hosting Logs: migrate to DataViews

oandregal opened the pull request:
- Depends on https://github.com/Automattic/wp-calypso/pull/96470 which needs to land first.
- Fixes https://github.com/Automattic/wp-calypso/issues/96793

## Proposed Changes

Updates the existing log table list to use DataViews

## Why are these changes being made?

We want to consolidate the listing interactions across our products into a single experience powered by DataViews.

## Testing Instructions

```sh
yarn && yarn start
```

- Visit `http://calypso.localhost:3000/sites`.
- Select an atomic site, which will open the preview panel.
- Go to the "Logs" tab.
- Verify that the table listing is using DataViews and works as expected.

## Screenshots

| | Modal with all fields | Filters in use |
| --- | --- | --- | 
| PHP | <img width="1998" alt="Screenshot 2025-03-04 at 11 57 19" src="https://github.com/user-attachments/assets/b46155a0-4489-490f-90a5-53ae013643e3" /> | <img width="1998" alt="Screenshot 2025-03-04 at 11 57 47" src="https://github.com/user-attachments/assets/477a08d2-c6d3-4bad-8aa4-a53b54def67b" /> |
| WebServer | <img width="1998" alt="Screenshot 2025-03-04 at 11 58 03" src="https://github.com/user-attachments/assets/265c68d7-dd14-462e-a1c0-ee29f323a9b9" /> | <img width="1998" alt="Screenshot 2025-03-04 at 11 58 24" src="https://github.com/user-attachments/assets/952d4497-bfaf-44db-b44f-ad57f070f28b" /> |

Date filter

<img width="1998" alt="Screenshot 2025-03-04 at 12 01 33" src="https://github.com/user-attachments/assets/2e5dacbc-e5ad-4f04-8413-efb7dc39c780" />


## Upstream issues reported

- https://github.com/WordPress/gutenberg/issues/69185
- https://github.com/WordPress/gutenberg/issues/69193
- https://github.com/WordPress/gutenberg/issues/69194
- https://github.com/WordPress/gutenberg/issues/69265
- https://github.com/WordPress/gutenberg/issues/69274
- https://github.com/WordPress/gutenberg/issues/69294

## Follow-ups

- Remove all code for the alternate route @fushar

oandregal committed:
Scaffold SiteLogsDataViews

oandregal committed:
Add header

oandregal committed:
Add basic logType toggle

oandregal committed:
Improve loading state

oandregal committed:
TypeScript

oandregal committed:
Add date filters

oandregal committed:
Better empty state

ntsekouras commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,239 @@ export const SiteLogs = ( {
+	// - DataViews: address the "show more" interaction.
+	// - DataViews: spacing left/right.
+
+	const { __ } = useI18n(); // Can we use translate instead?
```
How about importing from `@wordpress/i18n`? What the difference between importing, using `translate` or `useI18n`? 🤔

oandregal replied:
No idea why we use one or the other (there's also a `translate` call 🤷 ). Right now, I'm just trying to make things work following what the existing component chain does. Later, I'll dig into the details.

alshakero replied:
@wordpress/i18n is non-Reactive. It works in most cases, but if you change the locale or download language files after rendering, without refreshing the page, the UI won't refresh. Whereas `translate = useTranslate()` from `calypso-i18n` and `@wordpress/react-i18n` are both Reactive and will rerender the consumer when anything related to translations changes.

youknowriad replied:
As discussed with @tyxla and others, It's seems now there's more pros to using `@wordpress/i18n` than `calypso-i18n`. Should we be doing a P2 post on this to get everyone on the same page and understand the long term plan / migration...

tyxla replied:
Just to cnfirm, as @alshakero mentioned, In React context, we should be using `{ useI18n } from '@wordpress/react-i18n'` instead of `@wordpress/i18n` to benefit from the changes in locale or async loaded translation chunks. `@wordpress/i18n` is preferred only in non-React context. I did talk about it [here](https://github.com/Automattic/wp-calypso/pull/98476#discussion_r1920127424).

chriskmnds replied:
> As discussed with @tyxla and others, It's seems now there's more pros to using @wordpress/i18n than calypso-i18n. Should we be doing a P2 post on this to get everyone on the same page and understand the long term plan / migration...

Interesting. Can you point me to where these discussions took place. I was/am under a different impression, also judging from the sentiment I got in a similar discussion pxLjZ-9Eg-p2

I've recently dealt with dozens of translation issues primarily stemming from devs using a variety of i18n formatters, either not having the same locale or even the right locale internally. It explicitly concerned numbers, but numbers are just as important. Having all these packages available for similar (but not identical) purposes isn't serving the goal. 

As far as numbers go, I am unifying everything behind a single method: pxLjZ-9Ji-p2 And that method currently lives in `i18n-calypso`. I am not even sure if I will pull it out right now, as the alternatives aren't exactly better. `@wordpress/i18n` isn't reactive, `@wordpress/react-18n` is an extremely minimal package, `@automattic/i18n-utils` doesn't deal with internal state/locale definition, etc. 

I would 100% urge everyone, for now, to stick with i18n-calypso in Calypso until we work out the replacement (if we ever do). Just "using another" without proper migration isn't helping the code, and, judging from what I've seen the past couple of weeks, it's not helping the user either (and it slipped from everyone's eyes for a long time).

tyxla replied:
I may be missing something, but I don't understand why we need to be bound to `i18n-calypso` for number formatting. Yes, that's where the old utility function was before, but is that a good enough reason? What prevents us from moving closer to step 4 of your original proposal, and instead of having to bind number formatting to `i18n-calypso`, build it separately, and just allow it to take a locale in addition to the desired format? If we can make number formatting independent, then we would be able to use it with both `i18n-calypso` and `@wordpress/i18n` and `@wordpress/react-i18n`?

chriskmnds replied:
Well we are a couple of dozens of bugs ahead of my original proposal, all/most stemming from locale issues. Some similar situation exists (but not issues per se) with the JP formatter.

In my ideal scenario right now, `numberFormat` works as `translate`, with internal locale state. My fear with the alternative, of passing a browser-safe locale, is it won't be effective. People will insist on passing ant locale to it.

So internal locale state handling + optional locale param to encompass JP is what I'm shooting for. Make (keep) it stupid easy for most cases, expansive for others.

tyxla replied:
Got it. In any case, separating the browser-safe locale retrieval from `i18n-calypso` is something we can solve in the future. It's not a blocker for this PR, since all the works happens in Calypso's client and using `i18n-calypso` is still fine.

chriskmnds replied:
Yes, we'll come back to it. Not meaning this as a blocker. It just caught my attention, fighting with similar things.

A separate package imported into `i18n-calypso` (to wrap the calls with its locale state) might be right. wp-calypso will continue using it from there, never import it separately. There's also a lot going on around `setLocale`, not sure what it would mean to abstract it out somehow instead.

oandregal replied:
Consolidated everything to `useTranslate` in https://github.com/Automattic/wp-calypso/pull/99056/commits/e0bf787950ebbc5a0bc1e2a255cb0996c0750736

ntsekouras commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,239 @@ export const SiteLogs = ( {
+		setView( ( oldView ) => ( {
+			...oldView,
+			...newView,
+			type: 'table' as const,
```
Do we need this?

alshakero replied:
Yes, if you don't add `as const`, TS will view `type` as a mutable `string` and dataviews expect strictly `table`, `grid` and two other values I can't recall.

oandregal committed:
Sort

oandregal committed:
Pagination

oandregal committed:
Total items cannot be undefined

oandregal committed:
Disable search

oandregal committed:
Fields: names and sorting

alshakero commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
+
+	const { __ } = useI18n(); // Can we use translate instead?
+
+	const moment = useLocalizedMoment();
+	const getLatestDateRange = useCallback( () => {
+		const startTime = moment().subtract( 7, 'd' );
+		const endTime = moment();
+		return { startTime, endTime };
+	}, [ moment ] );
```
I don't want to make your life harder. But@wordpress/dataviews is 340KB gzipped and moment is 73KB gzipped. It would be amazing if we follow up with some performance considerations (eg async loading, not using moment, etc..).

alshakero commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
+			...newView,
+			type: 'table' as const,
+		} ) );
+	useEffect( () => {
+		setView( ( view ) => ( {
+			...view,
+			sort: {
+				field: getSortFieldForLogType( logType ),
+			},
+			fields: getVisibleFieldsForLogType( logType ),
+		} ) );
+	}, [ logType ] );
```
From what I see here, having `view` as a state is making your life harder. Maybe make all its dependencies states but the view itself can be a constructed in the end before returning.

```ts
const view = {
   type: 'table' as const,
   sort: { field: getSortFieldForLogType( logType ) },
   fields: getVisibleFieldsForLogType( logType ),
}
```

alshakero commented on client/sites/tools/logs/index.tsx:
```diff
@@ -1,11 +1,20 @@
+import { DataViews } from '@wordpress/dataviews';
 import { sprintf } from '@wordpress/i18n';
 import { useI18n } from '@wordpress/react-i18n';
+import { translate } from 'i18n-calypso';
```
I would use `useTranslate` instead. See why [here](https://github.com/Automattic/wp-calypso/pull/99056#discussion_r1936947493).

mcsf replied:
I understand the question of preferring the hook over the global `translate`, but I also see concurrent usage of `useI18n` and `useTranslate`; how does one choose one over the other?

tyxla replied:
Using `translate()` from `i18n-calypso` is not preferred since it is not reactive and changes to the locale or newly loaded translations will not be reflected. This isn't about `useI18n` at all.  As Omar recommended and gave context, `useTranslate()` from `i18n-calypso` is what we should be using here. You could also use `useI18n`, too, and there are very minor downsides which I've talked about [here](https://github.com/Automattic/wp-calypso/pull/98476#discussion_r1920127424).

mcsf replied:
> This isn't about `useI18n` at all.

No, I understood all that. This is was a follow-up question about the two different packages (btw, thanks for that link explaining it). Did my phrasing make it seem like I was asking about using the non-reactive `translate`?

tyxla replied:
> Did my phrasing make it seem like I was asking about using the non-reactive translate?

No, but I felt compelled to explain just in case, because it was a comment under a line that imports `translate` from `i18n-calypso`.

alshakero commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
+						<span>{ translate( 'From' ) }</span>
+						<DateTimePicker
+							className="site-logs-toolbar__datepicker"
+							id="from"
```
Do we need the id if it's wrapped by the label?

alshakero commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
+						<span>{ translate( 'To' ) }</span>
+						<DateTimePicker
+							className="site-logs-toolbar__datepicker"
+							id="to"
```
+1

alshakero commented on client/sites/tools/logs/index.tsx:
```diff
@@ -251,3 +263,257 @@ export const SiteLogs = ( {
 };
+
+const EMPTY_ARRAY: Array< any > = [];
+const useDataLogs = ( {
```
I would put this in a separate file.

alshakero reviewed (commented):
Hi everyone! Just a drive by review trying to help and learn about dataviews.

oandregal committed:
Filter by severity

oandregal committed:
Filter by status

oandregal committed:
Filter by request_type

oandregal committed:
Field labels: translate

oandregal committed:
Labels

oandregal committed:
Severity: add badge

oandregal committed:
Request type: add badge

oandregal committed:
Style badges properly

oandregal committed:
Fields: format dates

oandregal committed:
Control cell overflow

oandregal committed:
Add download logs button

oandregal commented:
Cross-posting status from p5j4vm-51h-p2#comment-7495.

**Current status as of January 31st**: all basic interactions are working.

- The only major interaction missing is “expand row to read more”. The DataViews native approach would be implementing it via a modal action.
- It needs style tweaks and fixes (spacing, date header, empty states for filters, etc.).
- In terms of data filters & sorting, there’s parity with the existing implementation. The current endpoint doesn’t support sorting by fields other than the timestamp, searching by text, filtering by multiple values (e.g., “severity is any of: …”), or adding more filters (e.g., filter by kind or name) — some of these would be valuable, but I consider them follow-up PRs.
- Code-wise, I just made it work. I’ve reused as many pieces as I could from the current implementation, but it’s not ready — I’ll address this next while design feedback comes.

oandregal committed:
TypeScript

oandregal committed:
Type fields

matt-west commented:
Nice work so far @oandregal!

### Spacing

It sounds like you’re already aware of this, but in the interest of being comprehensive, there are a couple spacing issues that need fixing:

* The DataViews table should span the entire width of the content page. There’s currently some left + right padding.
* The padding above the Logs title in the header is also inconsistent with other pages.
* 

### Auto-refresh

### Entry details behaviour

matt-west commented:
Nice work so far @oandregal!

## Spacing

It sounds like you’re already aware of this, but in the interest of being comprehensive, there are a couple spacing issues that need fixing:

* The DataViews table should span the entire width of the content page. There’s currently some left + right padding.
* The padding above the Logs title in the header is also inconsistent with other pages.

## Auto-refresh

The auto-refresh option is missing. Is it possible to place this in the toolbar above the table?

![image](https://github.com/user-attachments/assets/14946f13-6762-41d5-b5f2-3beab95133cd)

## Entry details behaviour

I’ve explored a few different ways we can show the details for a specific log entry. I suspect the modal approach will be the easiest to implement short-term as it doesn’t require changes to DataViews. I’m quite taken by the drawer approach though. What do you think?

### Modal
We could add an action icon to the row that triggers a modal for viewing the full details related to a log entry.

![image](https://github.com/user-attachments/assets/76668844-1049-4bce-9e48-9d8b16bce96d)


### Drawer
In this approach clicking a log entry would pop-up a drawer at the bottom of the content pane with the full details. Users could resize and close this drawer as they wish. (Sort of like browser dev tools)

Adding the drawer starts to make the content pane feel really crowded. Some of this will be relieved with the upcoming changes that simplify the site header and hide the sites list though.

![image](https://github.com/user-attachments/assets/9b05a6d4-fb68-47b6-8904-5faa996e9abc)


### Dropdown
This approach restores the current behaviour where selecting a log entry expands the row to display the full details. A nice advantage of this approach is that you can expand and compare multiple entries at the same time.

![image](https://github.com/user-attachments/assets/976ed5bf-5868-4703-941b-77df02c299f6)


## Educational tip

The filter icon above the table can be easily missed, so I think it warrants spotlighting that some of the filters have moved. We’ve used a similar component before so hopefully there's something you can reuse.

<img width="805" alt="Screenshot 2025-02-04 at 13 36 45" src="https://github.com/user-attachments/assets/f951a296-4491-4511-80c1-05d9582eccc4" />


## Cleaning up filters

We current have filters spread over multiple levels which is not ideal. I think this is a limitation of current DataViews capabilities, but just wanted to call it out.

<img width="1053" alt="Screenshot 2025-02-04 at 13 38 40" src="https://github.com/user-attachments/assets/96b0c740-4e12-4363-a1b0-3353008753e6" />

In the future it would be nice if we could integrate the date pickers into the toolbar to clean up the filters. I don't think DataViews supports this at the moment though. (Please correct me if I’m wrong!)

<img width="1037" alt="Screenshot 2025-02-04 at 13 40 24" src="https://github.com/user-attachments/assets/5cbbd24e-bfe3-4b80-9c20-33d544ce7556" />

oandregal committed:
Extract useFields and types

oandregal committed:
Rename SiteLogsTab type to LogType

oandregal committed:
Move types to existing place: use-site-logs-query

oandregal committed:
Extract useView

oandregal committed:
Extract useData

oandregal committed:
Fix field types

oandregal committed:
Provide a getItemId function

oandregal committed:
Clean up

oandregal committed:
Remove local state for date and use URL query instead

oandregal committed:
Add support for multiple values

oandregal committed:
Filters: multiple values for request_type and status + remove unnecessary All value

oandregal commented:
This week's progress was focused on looking at data flow/endpoints:

- Existing filters now support selecting multiple values.
- Pagination needs work: DataViews only supports per-page pagination but the endpoint/data storage only supports previous/next page.

| DataViews | Current experience |
| --- | --- |
| <img width="298" alt="Screenshot 2025-02-07 at 14 37 03" src="https://github.com/user-attachments/assets/5825ef05-a1e6-40be-a9ac-4587a955e494" /> | <img width="298" alt="Screenshot 2025-02-07 at 14 37 23" src="https://github.com/user-attachments/assets/36802eba-079a-4586-940a-e1b85f155273" /> |

The client requests data to some endpoints that end up triggering an ElasticSearch request. ElasticSearch does offer [filtering the dataset a per-page](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-search.html#search-search-api-example) via from and size parameters, but only for the first 10k items. Beyond that, you have to use the search_after [parameter](https://www.elastic.co/guide/en/elasticsearch/reference/current/paginate-search-results.html#search-after) that only allows you to request the previous and next page. The atomic and wpcom endpoints only support the search_after parameter.

On the other hand, DataViews doesn't support disabling the per-page navigation (so users would only see the previous/next links).

How do we reconcile this? A few alternatives:

- Do not use DataViews pagination and render a different control in the footer that only displays previous/next links.
- Prepare a patch to configure DataViews pagination so it doesnt' to display the pages, only the previous/next links.
- Prepare a patch to the endpoints, so they allow filtering the dataset per-page using from and size. This will be limited to 10k anyway.

oandregal committed:
ServerLog: format date

oandregal committed:
Fix padding left

oandregal committed:
Wrap long content

oandregal committed:
Set date/timestamp as titleField

oandregal committed:
Better spacing for table columns

oandregal committed:
Add copy action

oandregal committed:
Fix types

oandregal committed:
Add titleField to effect

oandregal committed:
Enqueue view update without effect

oandregal committed:
Address spacing issues in header

oandregal committed:
Add auto-refresh

oandregal committed:
Better naming

oandregal committed:
Clear filters upon logType changes

oandregal committed:
Add back track event

oandregal committed:
Simplify time changes logic

oandregal committed:
Communicate download is in progress

oandregal committed:
Track severity

oandregal committed:
Add boolean props per filter value for segmentation

oandregal committed:
Loading state: only use skeleton on first load and use empty state from dataviews

oandregal committed:
Use empty data & pagination when it is fetching

oandregal committed:
Use same visual loading mechanism on initial load and subsequent fetches

oandregal committed:
Maintain existing data until new comes in to minimize table layout shifts

oandregal committed:
Clear cached data when changing logType

oandregal committed:
Hide per-page navigation in dataviews

oandregal committed:
Add upstream ticket

oandregal committed:
Disable auto-refresh when switching pages

oandregal committed:
Disable auto-refresh on date filter change

oandregal committed:
Add missing fields from endpoint as hidden

oandregal committed:
Add renderer filter

oandregal committed:
Add cached filter

oandregal commented:
This week progress (see also p5j4vm-51h-p2#comment-7626 ):

- All feedback from design & the self-serve call has been addressed.
- The screen is feature-ready, and the **next steps** are:
  - Testing.
  - Clean up code (remove old implementation, consolidate, etc.) and focus on the code review.
  - Figure out how to add a tooltip to highlight that some filters have moved. DataViews doesn't allow us to add extra behavior to the filter button.
  - Add a message/tooltip when the copy action is triggered. What's calypso mechanism for this?

- **Improvements** over the existing version:
  - Filters: all filters allow multi-selection. In web logs, there are two additional filters (cached, renderer).
  - Fields: users can see all existing data from the endpoint, but the new fields are hidden by default. This is the same data users can download.
    - New hidden fields in PHP logs: “atomic_site_id”.
    - New hidden fields in web logs: “http2”, “http_user_agent”, “http_version”, “http_x_forwarded_for”, “renderer”, “request_completion”, “request_time”, “scheme”, “type”, “user_ip”.
- Auto-refresh: it now works with filters. It's only disabled if the user clicks pagination or date filters, while in the existing version it'd be also disabled upon selecting any other filter.
- Actions: new “copy action”. It copies the request URL in web server logs, and the message in PHP error logs.
- Sorting: data can be sorted in ascending or descending order. The existing version only allows descending.
- Wrap messages (PHP logs) and request URL (web logs) so users can see multiple at once.
- DataViews upstream issues reported:
  - Pagination: [can't disable per-page pagination](https://github.com/WordPress/gutenberg/issues/69193). 
  - Loading status: [can't set visual indicator (opacity)](https://github.com/WordPress/gutenberg/issues/69194).

matt-west commented:
Nice work @oandregal! Thanks for continuing to improve this.

## Copy action

We only need to show the copy button when a user hovers/focuses the row on desktop.

> Add a message/tooltip when the copy action is triggered. What's calypso mechanism for this?

I agree. We could use a [Notice](https://wpcalypso.wordpress.com/devdocs/design/notice) that appears in the top right of the screen and automatically dismisses after a couple seconds. It feels a bit heavy handed for a copy action. We typically update the text in a copy button to "Copied!" when it’s clicked, but that doesn’t really apply here given that it’s an icon button. (@jameskoster do you have any thoughts on a better way to handle this?)

![image](https://github.com/user-attachments/assets/b62417fb-cb18-4fe5-9cfb-a8adadf88353)


## Table spacing

The left padding for the table headers is slightly off.

<img width="753" alt="Screenshot 2025-02-14 at 09 47 46" src="https://github.com/user-attachments/assets/05617d80-6abf-4941-a48b-946c3d4b37ed" />


## Pagination

Can we pin the pagination bar to the bottom of the content pane? The table rows should then scroll beneath it.

Here’s an example from the Templates page in Core for reference:

<img width="1748" alt="Screenshot 2025-02-14 at 09 50 12" src="https://github.com/user-attachments/assets/79b6695d-d328-4f29-9160-9af43744761c" />


## Viewing the full log details

I think we still need an easy way to display all the fields for a log entry at once. In the short-term, the [modal](https://github.com/Automattic/wp-calypso/pull/99056#issuecomment-2634028888) is probably the most realistic option for that. We can add an action icon to trigger that, and maybe even link the timestamp in the table too.

Wrapping the text so that it it’s not truncated has helped a lot, but at the moment if you want to see all the fields for a log entry you need to toggle them on manually in the appearance settings. Then next time you refresh the page, the appearance settings are reset. It’s not a great experience.


## Mobile: Min-width for text-heavy columns

The `request url` and `message` columns become really difficult to read on mobile. Can we set a min-width on those columns to ensure they stay readable. It’s fine if that means the table has to scroll horizontally.

<img width="393" alt="Screenshot 2025-02-14 at 09 46 00 copy" src="https://github.com/user-attachments/assets/19d3d87e-1256-4b4f-ab2a-fabc2e7ebf76" />


## Mobile: Stack filter labels

Let’s clean up the layout of the date and log types filters on mobile.

| Header | After |
|--------|--------|
| <img width="393" alt="Screenshot 2025-02-14 at 09 46 00" src="https://github.com/user-attachments/assets/4d468a9a-ef58-4c4a-a83d-afc5614a6ea6" /> | <img width="414" alt="Screenshot 2025-02-14 at 10 30 21" src="https://github.com/user-attachments/assets/caec0098-73b5-4c74-8d53-cd73666d6200" /> |

jameskoster commented:
> We typically update the text in a copy button to "Copied!" when it’s clicked, but that doesn’t really apply here given that it’s an icon button. (@jameskoster do you have any thoughts on a better way to handle this?)

I think updating the button label is actually a bit of an a11y anti-pattern—it will only be announced on refocus (which may not happen), and generally button labels should reflect the on-click action. I think a snackbar is appropriate here, and would be consistent with the 'Copy all blocks' action in the editor.

matt-west commented:
Thanks for the input @jameskoster. We’re thinking about transitioning these notices to use the Snackbar in future anyway, so make sense to use it here.

matt-west commented:
@oandregal Would it be possible to use the date picker from Jetpack Stats here? It’s a lot more compact than the current split fields solution.

<img width="396" alt="Screenshot 2025-02-14 at 16 08 12" src="https://github.com/user-attachments/assets/fafb72d0-b3b5-4225-8864-576f44fcaf54" />

mcsf committed:
Show success/error notices upon copying a Log entry

oandregal committed:
Remove no longer necessary code

oandregal commented on client/sites/tools/logs/index.tsx:
```diff
@@ -1,253 +1,304 @@
-}
-
-export const SiteLogs = ( {
+export const SiteLogsDataViews = ( {
```
Despite what the diff suggest, SiteLogsDataViews is a brand new implementation based on DataViews, so it's best to ignore the diff.

oandregal committed:
Use existing DateControl instead of custom one

oandregal committed:
Better scope CSS rules

oandregal committed:
Remove unnecessary CSS

oandregal committed:
Improve styles

oandregal committed:
Make fields content wrap

oandregal commented:
Removed all the unnecessary code and the PR is now at +805 −1,778 LOC. The code structure is mostly what it needs to be and I welcome feedback on this. I've noticed that some small parts of the removed CSS need to be brought back (badges, for example).

Switched to the DateControl component used by the stats and this is how things look:

https://github.com/user-attachments/assets/58854e65-0355-43ea-8d8f-720d9fd6fc75

oandregal committed:
Fix badge styles

jameskoster commented:
Is there a reason not to use the filter UI for these?

<img width="719" alt="Screenshot 2025-02-20 at 09 42 32" src="https://github.com/user-attachments/assets/dac83b61-761c-455f-bbfd-69855f82dbd8" />

matt-west commented:
@jameskoster The PHP error/Web server selector needs to be distinct as it significantly changes the dataset. You can’t combine both types of logs in one table.

I’d love for us to integrate the date filter into the DataViews filter UI, but I don’t think this is supported. (Please correct me if that’s wrong!)

oandregal committed:
Add new modal action for PHPLogs and ServerLogs that displays all fields

oandregal commented:
Added a "View log details" action: it opens a modal that displays all the fields for that particular log type.

https://github.com/user-attachments/assets/ff398088-659e-4ef6-a5b4-05d47e900112

oandregal committed:
Mobile screens: logType filter

oandregal committed:
Set min width for long column fields

oandregal committed:
Mobile screens: date filter

jameskoster commented:
> The PHP error/Web server selector needs to be distinct as it significantly changes the dataset. You can’t combine both types of logs in one table.

I see! I guess it's not something for this PR but the position of the UI and navigational flow kind of makes it look like a filter.

> I’d love for us to integrate the date filter into the DataViews filter UI, but I don’t think this is supported. (Please correct me if that’s wrong!)

Yeah date filtering isn't natively supported, but would it be possible to implement this so that it that utilises the dataviews filter UX, rather than this separate UI? Perhaps in a way that can be contributed back later?

oandregal commented:
> Yeah date filtering isn't natively supported, but would it be possible to implement this so that it that utilises the dataviews filter UX, rather than this separate UI? Perhaps in a way that can be contributed back later?

No, we can't. But I proposed submitting a few couple of patches upstream so this, among other things, is extensible by consumers :) pgle0O-3F-p2

oandregal commented:
@matt-west I think I've addressed all your feedback so far, is there anything else you'd like to see here?

---

> The DataViews table should span the entire width of the content page. There’s currently some left + right padding.

This is the current status: the table grows to the max width defined by the `hosting-dashboard-item-view__content` class, which is the same across all tabs. @matt-west I'm not sure if there was a different spacing before, but wanted to confirm that the current state is ok.

https://github.com/user-attachments/assets/75ec557c-aefc-4527-b0a3-53670389d82e

> The filter icon above the table can be easily missed, so I think it warrants spotlighting that some of the filters have moved. We’ve used a similar component before so hopefully there's something you can reuse.

Other than the fields and its headers, DataViews doesn't expose mechanisms for consumers to hook external components to. I need to look into this one, perhaps hooking into a nearby component (Log header, table header) would do.

> The left padding for the table headers is slightly off.

This seems to be a DataViews thing that also happens in every other DataViews dotcom instance. Not sure if this is an artifact of the table header being a button or the lack of multi-selection, but I've created an upstream issue at https://github.com/WordPress/gutenberg/issues/69274

| core dataviews | dataviews in dotcom sites | 
| --- | --- |
| <img width="689" alt="Screenshot 2025-02-21 at 11 34 36" src="https://github.com/user-attachments/assets/6ba940e6-bb01-487f-97e9-0e894859ae5d" /> | <img width="102" alt="Screenshot 2025-02-21 at 11 37 31" src="https://github.com/user-attachments/assets/4e4d49b0-a87a-482c-8efa-23f6eaaf8940" /> |

> Can we pin the pagination bar to the bottom of the content pane? The table rows should then scroll beneath it.

How pagination is rendered is not configurable in DataViews.

Additionally, there's an issue with the lack of per-page navigation in the data source. What I did here was hiding the per-page control in DataViews and create an [upstream issue](https://github.com/WordPress/gutenberg/issues/69193) that I linked to the CSS hotfix. This kind of thing is problematic from an engineering point of view because it makes library updates harder and bound to break on updates.

I considered an alternative approach: hiding the pagination by telling DataViews there's no more pages and adding our own controls — like we do for the date filter or log type filter. This is also a hack, and it comes with more disjointed user experience (different pagination), so I thought the CSS hotfix was simpler for now.

> Viewing the full log details

Added a new non-primary action for this:

https://github.com/user-attachments/assets/84ec2d1d-af47-41df-9ac5-21108df0d81e

> Mobile

Pushed some changes for mobile and this is how it looks now:

https://github.com/user-attachments/assets/d018fc4f-db99-4d35-829c-477906b84c74

FWIW, I've also explored using the list layout instead of the table layout in mobile. I don't think it works well in this scenario. Other than some right/left spacing that would need to be fixed, note how it's very difficult to tell apart the last three fields (Request URL, HTTP Referrer, and HTTP Host):

<img width="413" alt="Screenshot 2025-02-21 at 12 03 33" src="https://github.com/user-attachments/assets/784c82d8-1778-49ad-b0c3-995f1ab6ba33" />

oandregal committed:
Translations: consolidate to useTranslate

oandregal committed:
Trigger a site settings request

matt-west commented:
Thanks for the update @oandregal!

> the table grows to the max width defined by the hosting-dashboard-item-view__content class, which is the same across all tabs

I think we should let logs go full-width (like the table on /sites).

> This seems to be a DataViews thing that also happens in every other DataViews dotcom instance

Thanks for logging an issue for that!

> pagination

Okay, sounds good. Thanks for exploring different solutions to that.

> Viewing the full log details

Can we link the timestamps in the first column to trigger the modal too?

> Mobile

I agree. The scrollable table is a lot more usable than the list view in this context.

chriskmnds commented on client/sites/tools/logs/hooks/use-actions.tsx:
```diff
@@ -0,0 +1,218 @@
+								<div className="site-logs-details-modal__field-title">{ translate( 'File:' ) }</div>
+								<div>{ item.file }</div>
+								<div className="site-logs-details-modal__field-title">{ translate( 'Line:' ) }</div>
+								<div>{ item.line }</div>
```
This looks like it's a number we are rendering in the HTML. Do we expect this to go over 1000? Would it make sense to parse with `numberFormat` so it gets properly localised? e.g. have the correct thousandth separators (`1,000` for EN, `1.000` for EL/DE, etc.)

oandregal replied:
That's a good thought. I've used `numberFormat` to formatting this and `body_bytes_sent`.

oandregal committed:
body_bytes_sent: format as a number

oandregal committed:
line: format as number

oandregal committed:
Exempt site logs from adhering to the max-width set for the rest of the tabs

oandregal committed:
Open modal on click the titleField

oandregal committed:
Fix timezone in PHP action modal

oandregal commented:
@matt-west I've made the table full-width and clicking on the "title field" (timestamp/date) opens the "log details" modal.

https://github.com/user-attachments/assets/f6d07705-9791-4ae0-8699-6dcf3666da8c

matt-west commented:
Excellent. Thanks @oandregal!

Can we update the modal to use the `large` size so it’s easier to read the log entry? The current width causes the text to wrap a lot which is a bit awkward.

<img width="274" alt="Screenshot 2025-02-24 at 13 52 45" src="https://github.com/user-attachments/assets/a50e19a4-f6dc-4514-8eaf-59e2bb5de7e9" />

The labels should also use the same styling we have for form labels. For example:

<img width="835" alt="Screenshot 2025-02-24 at 13 55 44" src="https://github.com/user-attachments/assets/0083114c-418b-4253-b50b-adfc4ac18e95" />

oandregal committed:
Tweaks to modals: use lowercase in title + use large size

oandregal committed:
Improve modal design

oandregal commented:
Done those changes to modals.

Note the modal opened by triggering the DataViews action "view details" is still using the "medium" size because that property can't be modified by consumers. Opened upstream issue at: https://github.com/WordPress/gutenberg/issues/69294

 They now look like this:

| PHPLog | ServerLog |
| --- | --- |
| <img width="875" alt="Screenshot 2025-02-24 at 17 32 03" src="https://github.com/user-attachments/assets/9a5b733c-ac17-47fd-9757-d6f1a946efe0" /> | <img width="886" alt="Screenshot 2025-02-24 at 17 31 51" src="https://github.com/user-attachments/assets/7c06455a-ec3c-4278-91d3-f28a6e9e9770" /> |

oandregal committed:
Fix types

matt-west commented:
Thanks @oandregal. Looks good 👍

oandregal committed:
Do not allow user to hide the titleField (timestamp/date)

oandregal committed:
Better naming

oandregal committed:
Refactor buildFilter to toFilterParams

oandregal committed:
Better context for translators

oandregal committed:
Better getItemId function

oandregal committed:
Fix TS issue

oandregal committed:
Reorganize date variables

oandregal committed:
Use TypeScript Enums instead of string literals

oandregal committed:
Use TypeScript Enums instead of string literals

ntsekouras commented:
I started testing this a bit and I think there is some kind of error when we change the `date` filter. I could reproduce and when it happens, then the table seems 'stuck' with the same set of data.

Is this happening for you too?

https://github.com/user-attachments/assets/442969a8-5072-45c3-83e4-0e261758c1fb

ntsekouras commented on client/sites/components/dotcom-style.scss:
```diff
@@ -314,7 +314,7 @@
 .wpcom-site .hosting-dashboard-layout.sites-dashboard:not(.domains-overview) .site-preview-pane {
 	.hosting-dashboard-item-view__content {
-		> * {
+		> *:not(.site-logs) {
```
Nit question: why added `not` instead of overriding `.site-logs`? Is it because order of loading?

oandregal replied:
This rule is where the max-width is set for all the tabs, so I thought adding it here was more future-proof. I suppose adding a new rule to set max-width to none in the logs tab could work, but it has the disadvantage that isn't collocated with this rule, so future changes to this can easily miss it.

ntsekouras commented on client/sites/tools/logs/index.tsx:
```diff
@@ -1,253 +1,332 @@
-	const [ severity, setSeverity ] = useState( () => {
-		return getFilterQueryParam( 'severity' ) || '';
-	} );
+			return 'unknown';
```
We should never be in this line, should we? If we can be, won't it affect many DataViews operations, like selection, etc?

oandregal replied:
Removed this dead code path at https://github.com/Automattic/wp-calypso/pull/99056/commits/aa50658120b0c2b95b88ed6ea0d92f209b39bb5d

ntsekouras commented on client/sites/tools/logs/hooks/use-view.ts:
```diff
@@ -0,0 +1,92 @@
+
+	if ( logType === LogType.WEB ) {
+		const cached = getFilterValue( view, 'cached' );
+		const requestType = getFilterValue( view, 'request_type' );
```
Nit: I guess you could iterate the array once and assign all values.

oandregal replied:
Applied the suggestion at https://github.com/Automattic/wp-calypso/pull/99056/commits/c9409cf90595983bb3aea72118477b0bc6edd7ef

ntsekouras commented on client/sites/tools/logs/index.tsx:
```diff
@@ -1,253 +1,332 @@
+			{ !! itemDetailsModal && (
+				<DetailsModal
+					item={ itemDetailsModal }
+					actions={ actions }
```
I don't really get why we need to pass the `actions` here. Shouldn't the reusable modal just render things based on the `item` and be used (imported) here and at the action that uses it too?

oandregal replied:
Applied the suggestion at https://github.com/Automattic/wp-calypso/pull/99056/commits/be8946dcb6412dc48eb9efef7809a6c78aefcfb9

ntsekouras reviewed (commented):
Thank you @oandregal ! That's a big PR 😄 

In general this seems to work quite well and took a better look at the DataViews stuff. My comments are minor, but some testing from other folks more familiar with the logs functionality, would be good to have.

Main concern is my [above comment](https://github.com/Automattic/wp-calypso/pull/99056#issuecomment-2685104614) for date filters.

oandregal commented:
> Is this happening for you too?

No, it doesn't 🤔 

https://github.com/user-attachments/assets/3b44f92a-7968-4925-a872-abb003936fc4

oandregal committed:
Remove dead code path

oandregal committed:
Extract modals for PHP and server

oandregal committed:
Refactor to use getFilterParamsFromView

oandregal committed:
Remove unnecessary prop

oandregal committed:
Add an id to the data based on the position of the record

ntsekouras reviewed (approved):
With the `id` change everything works well on my end. Thank you @oandregal!

oandregal commented:
> I started testing this a bit and I think there is some kind of error when we change the date filter. I could reproduce and when it happens, then the table seems 'stuck' with the same set of data.

Nik and I paired on this and it's now fixed. The issue he ran into was only reproducible in a specific site for PHP errors (but not for Web errors). We boiled it down to the way each record's ID was calculated: there's no unique ID for each record coming from the endpoint, so we need to derive one and the way it was done wasn't unique enough.

In https://github.com/Automattic/wp-calypso/pull/99056/commits/60dbe41f1e7b17e5dbf57d4f157bc97b908b0f1f I've switched to use the position of the log in the logs array as the ID. This is the along the same lines as the [existing implementation](https://github.com/Automattic/wp-calypso/blob/79e13c525f9ded11e331bfeaa3701ba9f30dc12a/client/sites/tools/logs/components/site-logs-table/index.tsx#L155) – which concatenates that key with some data coming from the item.

oandregal commented:
While testing, I've realized there's one difference from the existing implementation: the dataview filters won't update the URL (but the logType and dates would). 

- Before: `https://wordpress.com/site-logs/SITE/php?severity=User&from=1740476621&to=1740649421`
- After: `https://wordpress.com/site-logs/SITE/php?from=1740476621&to=1740649421`

I understand having the filters in the URL can be useful (some users may want to bookmark a particular log search), so I'll bring it back.

oandregal commented:
I've also taken a look at how the existing screen works in a different languages (Spanish, French). I've noticed that the filters are extracted for translation, but I don't think they should: someone who is looking at these errors would be familiar with what they represent and understands they come from standards used by machines.

The values in the "Severity" filter ("Gravedad"/"Gravité") are standard [PHP Error Levels](https://www.php.net/manual/en/errorfunc.constants.php). Those values in the table records cannot be translated because they are dynamically generated by the server, which creates some disonnance to readers.

| Spanish | French |
| --- | --- |
| <img width="1446" alt="Screenshot 2025-02-27 at 11 45 49" src="https://github.com/user-attachments/assets/58369e5b-161d-48cc-9141-8abce323a034" /> | <img width="1446" alt="Screenshot 2025-02-27 at 12 01 09" src="https://github.com/user-attachments/assets/af791905-c742-42ca-973f-fad1ac8841b9" /> |

The same happens in the case of the existing "Request type" filter ("Tipo de solicitud"/"Type de requête"), that expose [HTTP Request Methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods). I'd even argue this is a bit more special in that this is part of a protocol used my machines.

| Spanish | French |
| --- | --- |
| <img width="1446" alt="Screenshot 2025-02-27 at 11 46 06" src="https://github.com/user-attachments/assets/eab5d007-c1b9-4afa-acd9-5ddf1a7f29d1" /> | <img width="1446" alt="Screenshot 2025-02-27 at 12 16 54" src="https://github.com/user-attachments/assets/8486e46e-a3c4-42ce-94db-0d23f3a404b5" /> |

oandregal committed:
URL contains filter parameters

oandregal committed:
Minimize re-renders when users interact with the filters

oandregal commented:
I'd like to get some thoughts from @Automattic/lego here. There are two URLs that load the logs tab:

1. `http://wordpress.com/site-logs/SITE/php` this is where you end up if you navigate from "Sites (hosting screen) > Select a site > Click logs table".
2. `http://wordpress.com/sites/tools/logs/SITE/php` I don't know how to get to this screen, but in this PR it looks broken:

How do users get to that second URL? Why do we have both and can we remove the second one?

<img width="1041" alt="Screenshot 2025-02-28 at 10 43 10" src="https://github.com/user-attachments/assets/eb743e88-e212-4fd3-badd-94489eaf37e9" />

matt-west commented:
@oandregal I ran another test on this and picked up a couple minor things. Once these are resolved, I think we’re good to ship this.

### Center loader and empty state

If there are no entries in the table—or the table is loading—we need to center the content vertically within the content pane.

| Empty state | Loading |
|--------|--------|
| <img width="1356" alt="Screenshot 2025-02-28 at 09 38 12" src="https://github.com/user-attachments/assets/b8fc0eec-d1e2-482d-b681-a2c25802c732" /> | <img width="1973" alt="Screenshot 2025-02-28 at 09 39 53" src="https://github.com/user-attachments/assets/4353a4bb-2b64-490c-bcf1-3dd2121bd5d1" /> | 

### Date picker

The date picker looks out of place as it uses a different font-weight and border-radius to the other fields.

<img width="465" alt="Screenshot 2025-02-28 at 09 39 42" src="https://github.com/user-attachments/assets/d4409a9c-5469-4de7-ae6c-38592080340c" />

We need to fix this at the component level, so I think it’s okay to ship this as-is and we can address it in a follow-up PR.

fushar commented:
> http://wordpress.com/sites/tools/logs/SITE/php I don't know how to get to this screen, but in this PR it looks broken:

The `/sites/tools/logs/*` route was used for pbxlJb-6ye-p2, which is unfortunately canceled. We can ignore it for now and use the current `/site-logs/*` URL.

oandregal committed:
Fix conflicting styles: center loading/noresults + fixed dataviews footer

oandregal commented:
> If there are no entries in the table—or the table is loading—we need to center the content vertically within the content pane.

There was a conflict between dotcom styles & dataviews styles. I've fixed it by https://github.com/Automattic/wp-calypso/pull/99056/commits/2a6e6652ab51a5427e1952dce7f09679c957782b

This also fixes the something that went unnoticed: the DataViews footer (pagination) is now floating at the bottom of the viewport as it should be. For proxied automatticians pagination ends up a bit hidden by the floating thing for dev/staging, but this doesn't affect customers nor unproxied automatticians.

Before:

https://github.com/user-attachments/assets/b909b684-0029-46fc-8e20-492b5bf38750

After:

https://github.com/user-attachments/assets/86760ded-c84e-41d6-85a3-274c7b123076

@matt-west I'd like to give you time to see this change, so I'll wait for 🟢 to merge the PR.

> re: DateRange We need to fix this at the component level, so I think it’s okay to ship this as-is and we can address it in a follow-up PR.

👍 

Yeah, I agree. There're other improvements we could make to this component. Code-wise, consumers shouldn't need to provide presets unless they want to override the defaults. Interaction-wise, it's not clear when it'll set the start or the end date, no "clear" button, etc. However, it's been used for years in the stats and other parts of the product, so I don't think any of this should be a blocker for this PR.

> http://wordpress.com/sites/tools/logs/SITE/php I don't know how to get to this screen, but in this PR it looks broken:

I chatted with @fushar and all this code can be removed. Can be done in parallel to this PR, so not a blocker for landing.

oandregal committed:
Limit scroll to dataviews-wrapper

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17280285

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @oandregal, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `DATE & TIME (%(siteGsmOffsetDisplay)s)`
- `SEVERITY`
- `MESSAGE`
- `KIND`
- `FILE`
- `LINE`
- `ATOMIC SITE ID`
- `REQUEST TYPE`
- `HTTP STATUS`
- `BODY BYTES SENT`
- `CACHED`
- `HTTP HOST`
- `HTTP REFERRER`
- `HTTP/2`
- `User AGENT`
- `HTTP VERSION`
- `X-FORWARDED-FOR`
- `RENDERER`
- `REQUEST COMPLETION`
- `REQUEST TIME`
- `SCHEME`
- `TIMESTAMP`
- `TYPE`
- `USER IP`
- `Log details`
- `Copy message`
- `Copy failed`
- `View log details`
- `Copy request URL`
- `Date & time (%(siteGsmOffsetDisplay)s)`
- `Atomic Site ID`
- `HTTP Referrer`
- `HTTP2`
- `HTTP User Agent`
- `HTTP Version`
- `HTTP X Forwarded Port`
- `Renderer`
- `Request Completion`
- `Scheme`
- `User IP`
- `Select a date range`

Thank you in advance!

oandregal commented:
> Hi @oandregal, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:

Added a few screenshots that should cover all strings.

oandregal commented:
Follow-up https://github.com/Automattic/wp-calypso/pull/100763 to dispatch notices upon copying the request url for webserver logs.

oandregal commented:
I'm iterating over a few details (long messages and "open details" action) over at https://github.com/Automattic/wp-calypso/pull/100784 This is something that has been brought up in the project thread at p5j4vm-51h-p2#comment-7710

tyxla commented on client/sites/tools/logs/hooks/use-actions.tsx:
```diff
@@ -0,0 +1,90 @@
+							dispatch(
+								successNotice(
+									/* translators: notice shown upon copy of Logs entry */
+									translate( 'Copied' )
```
I was testing these. and a few things stood up for potential improvement:

* Can we be more clear about what we're copying? "Copied" seems a bit too short. I'd go with "Copied message to clipboard" or "Copied request URL to clipboard" similar to our messages in core.
* For PHP logs, we're copying the message. Is that enough? Could it be more helpful to copy the entire entry that includes the timestamp?
* For web server logs, we're copying just the request URL, I'd suggest being able to copy more of the entry.

oandregal replied:
I've addressed the copy suggestions at https://github.com/Automattic/wp-calypso/pull/100827

tyxla commented on client/sites/tools/logs/hooks/use-actions.tsx:
```diff
@@ -0,0 +1,90 @@
+							dispatch(
+								errorNotice(
+									/* translators: notice shown upon failed copy of Logs entry */
+									translate( 'Copy failed' )
```
How relevant/useful is this notice? I don't think it helps much. Seems like we fail silently in Gutenberg.

tyxla commented on client/sites/tools/logs/hooks/use-actions.tsx:
```diff
@@ -0,0 +1,90 @@
+					callback: async ( items: ( PHPLog | ServerLog )[] ) => {
+						const message = ( items[ 0 ] as PHPLog ).message;
+						try {
+							await navigator.clipboard.writeText( message );
```
I wonder if we considered reusing `useCopyToClipboard()` from core?

tyxla commented on client/sites/tools/logs/components/details-modal-php.tsx:
```diff
@@ -0,0 +1,59 @@
+			<div className="site-logs-details-modal__field-title">
+				{
+					// translators: %(siteGsmOffsetDisplay)s will be replaced with the timezone offset of the site, e.g. GMT, GMT +1, GMT -1.
+					translate( 'DATE & TIME (%(siteGsmOffsetDisplay)s)', {
```
I wonder why we preferred to go with uppercase strings. I'd strongly recommend these to be lowercase or sentence case, and then we could always rely on CSS for uppercasing.

oandregal replied:
This was a design request: the idea was to mimic [form controls style](https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-inputcontrol--docs).

tyxla replied:
I'm not sure we're talking about the same thing. I'm saying we could use lowercase / sentence case, but then achieve the same effect visually with `text-transform: uppercase`. 

My rationale is that uppercase strings can feel weird to translators.

matt-west replied:
Agree with using sentence case for the translated strings, then using CSS to transform to uppercase.

I believe this is how we handle form labels at the moment.

tyxla replied:
It's worth noting that this is a low-priority issue: it doesn't affect users, but rather, only translators.

oandregal replied:
> I'm not sure we're talking about the same thing. I'm saying we could use lowercase / sentence case, but then achieve the same effect visually with text-transform: uppercase.
> 
> My rationale is that uppercase strings can feel weird to translators.

Oh, I misunderstood your point. Upon reflection, I think there's a bigger reason to use CSS for uppercasing: make sure they are always uppercase, no matter what the translator uses (e.g., the translator can use lowercase instead of uppercase). I missed this point, so I'm glad you raised it! I'm preparing a PR to fix this in a bit.

oandregal replied:
PR at https://github.com/Automattic/wp-calypso/pull/100836

tyxla commented on client/sites/tools/logs/components/details-modal.tsx:
```diff
@@ -0,0 +1,39 @@
+
+	if ( logType === LogType.PHP ) {
+		return (
+			<Modal
+				title={ translate( 'Log details' ) } /* Same title as dataviews action modal */
+				onRequestClose={ onClose }
+				focusOnMount="firstContentElement"
+				size="large"
+			>
```
The `Modal` part is repeated, so we have 2 identical copies of this modal. We could have the conditional logic responsible only for rendering the inner modal contents. Unless there's a good reason for the duplication?

oandregal replied:
I just thought it was more readable that way, which I prioritize over writing less characters :)

tyxla replied:
I disagree, but leaving this to your judgment

tyxla reviewed (commented):
Great work on this one! 

I gave it some testing on a couple of my sites, and I am providing some feedback. Works really well overall, I guess we can polish the notices a bit. 

Let me know what you think!

tyxla commented on client/sites/tools/logs/hooks/use-fields.tsx:
```diff
@@ -0,0 +1,266 @@
+				enableSorting: false,
+			},
+			{
+				id: 'http_x_forwarded_for',
+				type: 'text',
+				label: translate( 'HTTP X Forwarded Port' ),
```
Just spotted this as I was reviewing #100836.

This seems like a mistake. Both are headers, but: 

* `HTTP_X_FORWARDED_FOR`: contains the IP address of the original client making the request
* `HTTP_X_FORWARDED_PORT`: specifies the port number that the original client used to connect to the proxy or load balancer.

I imagine you meant to name this "HTTP X Forwarded For", although this is quite a technical bit and we could as well keep the original header name (`HTTP_X_FORWARDED_FOR`).

oandregal replied:
That's an eagle eye, thanks! PR at https://github.com/Automattic/wp-calypso/pull/100931
